arrow-fx/src/main/kotlin/arrow/fx/internal/UncancellableMVar.kt

```diff
@@ -64,7 +64,7 @@ internal class UncancellableMVar<F, A> private constructor(initial: State<A>, pr
         var first: Listener<A>? = null
         val update: State<A> =
           if (takes.isEmpty()) {
-            WaitForTake(a, IQueue.empty())
+            State(a)
```

Comment:
I moved from `State(a)` to more explicit `WaitForTake` recently since I felt that `State` seriously hindered readability.

https://github.com/arrow-kt/arrow/pull/2023/files#diff-14d465adcba579f7d8df401c3f1b3af4L63